ServerCore/Pages/Events/Mailer.cshtml.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Events
+{
+    // Note: Mailer is restricted to admins because the number of messages we can send is a limited resource.
+    // This prevents author abuse/ignorance.
+    [Authorize(Policy = "IsEventAdmin")]
+    public class MailerModel : EventSpecificPageModel
+    {
+        public enum MailGroup
+        {
+            AllPlayers,
+            AllPrimaryContacts
+        }
+
+        [BindProperty]
+        public MailGroup Group { get; set; }
+
+        [BindProperty]
+        [Required]
+        public string MailSubject { get; set; }
+
+        [BindProperty]
+        [Required]
+        public string MailBody { get; set; }
+
+        public MailerModel(PuzzleServerContext serverContext, UserManager<IdentityUser> userManager) : base(serverContext, userManager)
+        {
+        }
+
+        public IActionResult OnGet(MailGroup group)
+        {
+            Group = group;
+            return Page();
+        }
+
+        public async Task<IActionResult> OnPostAsync()
+        {
+            if (!ModelState.IsValid)
+            {
+                return Page();
+            }
+
+            IEnumerable<string> addresses = Enumerable.Empty<string>();
+
+            switch(Group)
+            {
+                case MailGroup.AllPlayers:
```

Comment:
For PuzzleDay we may need to email individual teams (e.g. "Go to the live event that you forgot"). I can't immediately think of a case for Hunt though.

Reply:
Added the ability to mail a team.